compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/test/tiling.mlir

```diff
@@ -2058,3 +2058,258 @@ module attributes { transform.with_named_sequence } {
+    %tiled_att, %grid = transform.structured.tile_using_forall %0 tile_sizes [4, 128] : (!transform.any_op) -> (!transform.any_op, !transform.any_op)
+    transform.yield
+  }
+}
+// CHECK-LABEL: func @custom_op_tile_scalar_args(
+//  CHECK-SAME:     %[[ARG0:[a-zA-Z0-9]+]]: tensor<?x?xf32>, %[[ARG1:[a-zA-Z0-9]+]]: f32,
+//       CHECK:   scf.forall
+//       CHECK:     %[[SLICE:.+]] = tensor.extract_slice %[[ARG0]]
+//       CHECK:     iree_linalg_ext.custom_op
+//  CHECK-SAME:         ins(%[[SLICE]], %[[ARG1]]
+
+// -----
+
+func.func @custom_op_index_handling(%arg0 : tensor<?x?xindex>,
+    %arg2 : tensor<?x?xindex>) -> tensor<?x?xindex> {
+  %0 = iree_linalg_ext.custom_op {
+      indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>,
+                       affine_map<(d0, d1, d2) -> (d0, d1)>],
+      iterator_types = [#iree_linalg_ext.iterator_type<parallel>,
+                        #iree_linalg_ext.iterator_type<parallel>,
+                        #iree_linalg_ext.iterator_type<reduction>]}
+      ins(%arg0 : tensor<?x?xindex>) outs(%arg2 : tensor<?x?xindex>) {
+    ^bb0(%b0 : tensor<?x?xindex>, %b2 : tensor<?x?xindex>):
+      %1 = iree_linalg_ext.index 0 : index
+      %2 = linalg.generic {
+          iterator_types = ["parallel", "parallel", "reduction"],
+          indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>,
+                           affine_map<(d0, d1, d2) -> ()>,
+                           affine_map<(d0, d1, d2) -> (d0, d1)>]}
+          ins(%b0, %1 : tensor<?x?xindex>, index)
+          outs(%b2 : tensor<?x?xindex>) {
+        ^bb0(%bb0: index, %bb1: index, %bb2: index):
+          %2 = arith.muli %bb0, %bb1 : index
+          %3 = arith.addi %2, %bb2 : index
+          linalg.yield %3 : index
+      } -> tensor<?x?xindex>
+      iree_linalg_ext.yield %2 : tensor<?x?xindex>      
+  } -> tensor<?x?xindex>
+  return %0 : tensor<?x?xindex>
+}
+module attributes { transform.with_named_sequence } {
+  transform.named_sequence @__transform_main(%module_op: !transform.any_op {transform.readonly}) {
+    %0 = transform.structured.match ops{["iree_linalg_ext.custom_op"]} in %module_op : (!transform.any_op) -> !transform.any_op
+    %tiled_att, %grid = transform.structured.tile_using_forall %0 tile_sizes [4, 128] : (!transform.any_op) -> (!transform.any_op, !transform.any_op)
+    transform.yield
+  }
+}
+//      CHECK: #[[MAP:.+]] = affine_map<(d0, d1) -> (d0 + d1 * 4)>
+//      CHECK: func @custom_op_index_handling(%[[ARG0:[a-zA-Z0-9]+]]: tensor<?x?xindex>,
+//      CHECK:   scf.forall (%[[IV:[a-zA-Z0-9]+]],
+//      CHECK:     %[[SLICE:.+]] = tensor.extract_slice %[[ARG0]]
+//      CHECK:     iree_linalg_ext.custom_op
+// CHECK-SAME:         ins(%[[SLICE]]
+//      CHECK:       %[[NEW_INDEX:.+]] = iree_linalg_ext.index 0 : index
+//      CHECK:       %[[INDEX:.+]] = affine.apply #[[MAP]](%[[NEW_INDEX]], %[[IV]])
+//      CHECK:       linalg.generic
+// CHECK-SAME:           ins(%{{.+}}, %[[INDEX]] :
```

Comment:
Can you add an example where the a linalg.generic is nested within a linalg_ext.custom_op and uses linalg.index? I want to see if linalg.index gets changed accordingly.

Reply:
I can add an example, but want to make sure I understand what you expect. The `linalg.index` within the `linalg.generic` should not change